updating source files

cekees opened the pull request.

cekees committed:
updating source files

cekees commented on flexible_plate_dambreak_2D/dambreak.py:
```diff
@@ -399,7 +399,7 @@ def get_w(self):
     def calculate_init(self):
         self.last_F = None
-        self.calculate()
+        #self.calculate()
```
@Milad-Rakhsha iirc this call to calculate() occurs too early  in the initialization to have the pressure attached. I don't think it's actually necessary to call it here, so I just commented it out.

Milad-Rakhsha replied:
So I am getting the same segfault in the flexible plate and the cylinder drop examples, but you mentioned that you are already running the cylinder drop example. I will compare what you have vs what I have for that example and the would lead me to solve the segfault example in the flexible plate example as well. I believe that the two segfaults are essentially caused by the same issue

cekees replied:
To be clear, the only reason I changed the sdf calculation to the internal distance calculation was because of the speed. I could run it in either mode.

Milad-Rakhsha commented on cylinderDropIBM/twp_navier_stokes_p.py:
```diff
@@ -49,12 +49,17 @@
                                     particle_beta=1e5,
                                     particle_penalty_constant=1e6,
-                                    particle_sdfList=[obj_sdf_Calc],
-                                    particle_velocityList=[obj_vel_Calc])
```
Okay so I think the only difference between the model you ran for the cylinder problem and the ones I had here is that you calculate the sdf for rigid balls from the position/velocity of MBD solver, whereas I gradually had changed this to calculate the sdf for any shape via the particle_sdfList, particle_velocityList. This was needed because for the flexible plate example the sdf has to be computed that way.
Anyway, I had made some changes in the proteus source codes (unfortunately in the same commit as I did the merge with the latest proteus, so it is unclear what the actual change was) to be able to run these and they are available here :
https://github.com/Milad-Rakhsha/proteus
what I recall is that the bug fixes in one of the RANS*.h files I made was actually necessary to get the sdf calculator to work properly (I think there was an out-of-bound memory access that I had fixed).

cekees replied:
OK, sorry for the slow response. Are you saying that in your merge with the latest proteus main branch you are able to run the flexble plate?

Milad-Rakhsha replied:
sorry for the confusion, no what I meant was that I could run the flexible plate problem with the branch in the link above. However, the branch is not merged with the latest master. When I last merged it (2 years ago), I incorporated all the bug fixes with the merge commit so that is why it is pretty difficult to see what was part of the merge and what was the actual bug fixes I needed to have the sdf calculators working properly

Milad-Rakhsha replied:
Would it be straightforward to try to build this old branch in a current proteus conda environment? I am asking because most of the dependencies are installed within  the conda env

cekees replied:
I think  it should work. Most of the the time the changes to the dependencies only cause a couple of easy proteus changes, at most. (e.g. petsc and hdf5 api changes).

cekees committed:
fix flexible plate problems

- add '-std=c++17' in setup.py (this is required for eigen)
- update chrono matrix usage
- update to python3 syntax
- update to proteus 1.8.0 mprans classes (no RDLS3P anymore)

cekees commented on FSI_Validation/setup.py:
```diff
@@ -51,10 +51,10 @@
                              libraries=['ChronoEngine',
                                         'stdc++','m'],
-                             extra_compile_args=["-std=c++11"]),
+                             extra_compile_args=["-std=c++17"]),
```
@Milad-Rakhsha, this appears to have been the source of the problem I was having. More here:
http://eigen.tuxfamily.org/dox-devel/group__TopicUnalignedArrayAssert.html

I'm running the FSI_Validation problem now.

/cc @DanNegrut

Milad-Rakhsha replied:
That's great thanks for figuring this out! I think back when I was working on this Chrono was not employing Eigen so using c++11 standards was just fine, but I am glad that everything looks to be working fine again!

cekees commented:
![vel 0016](https://user-images.githubusercontent.com/613654/127328791-1b442f85-9aa8-45b5-abfc-caf42c4cabb0.png)

The results are looking OK.  Most of the time is being taken in the Chrono step  (80s/step versus 3s/step for the Proteus step). Not that big of a deal, I'm sure it needs to be profiled, and I think the current build I'm using had Eigen built in debug mode and likely some other sub-optimal choices.
 
```
[   68534] Min_dt_controller setSubsteps tList=[0.079889361406125] self.t_model=0.079889361406125 self.substeps= [0.079889361406125] 
time/dt before chrono calculation: 0.07980256110062266 8.627680374361094e-05
[   68534] Calling chrono with dt 8.627680374361094e-05
skipping node 325 in force balance.
skipping node 326 in force balance.
skipping node 327 in force balance.
skipping node 328 in force balance.
skipping node 329 in force balance.
skipping node 645 in force balance.
skipping node 646 in force balance.
skipping node 647 in force balance.
skipping node 648 in force balance.
skipping node 649 in force balance.
totalForce to chrono dA=0.000013 (0.368837, 0.135112, 0.000116) 
tip node = 0.129179,0.005420,0.000000
DoStepChronoSystem with dt= 0.000086
 chrono time : 0.079803
[   68614] System time step t= 7.98894e-02, dt= 8.68003e-05
```

Milad-Rakhsha committed:
Merge branch 'miladrakhsha' into cekees
